### ISSUE TITLE
write_graphml - key error on non-existent key type

### ISSUE THREAD

**Issue by MikeB2019x** [networkx/networkx#5071]
### Current Behavior
I'm trying to export a large graph (144k nodes, 435k edges) to the GRAPHML format.  I tried exporting a subset initially and that went well.  Next I tried to export the whole graph. The export fails with the error below.  The challenge is that I don't understand what that error refers to; there are no nodes or edges with a 'key' that is a dict.  All the nodes and edges that have attributes assigned had them created using `G.nodes[].update()` or `G.edges[],update()`.

![image](https://user-images.githubusercontent.com/49003263/132545353-6589bf4d-a899-43c2-997f-4ba5c11c08f8.png)

### Expected Behavior

The entire graph exports to the GEXF format without problems (though I can't use that format).   That suggested to me that the graph was correctly constructed and so I expected the export to GRAPHML to also work.

### Steps to Reproduce

I can reproduce the error with the following.  Note that as in my graph the _**keys**_ are all 'int' or 'str'.  Some _**values**_ are 'dict' but none of the keys are. 
```
test_G = nx.Graph()
test_G.add_node(50509)

attributes = {'label': '55509', 'type': 'document', 
             1: {'amount': 42.8, 'currency': 'EUR','description': 'blueprint'}, 
             2: {'amount': 104.12, 'currency': 'EUR', 'description': 'blueprint'}, 
             3: {'amount': 116.0, 'currency': 'EUR', 'description': 'blueprint '}
             }

test_G.nodes[50509].update(attributes)
print(test_G.nodes(data=True))
nx.write_graphml(test_G,'error_G.graphml')
```

### Environment

OS: macOS (Catalina)
Python version: 3.8
NetworkX version: 2.6.2

### Additional Context
It looks like the structure of the attributes is causing the problem ie. 'dict' as values but I thought that was an acceptable construct.

**Comment by dschult** [comment]
Unfortunately, GraphML doesn't support dicts or lists as attribute values.  You can convert them to strings, but then you have to convert them back to dicts when you read them.  The reason the error is a KeyError is that the code is looking up the type of the value (a dict) in the dictionary of types that our writer knows how to handle. 

The poor error warning should be fixed by a recently submitted pull request #5058

**Comment by MikeB2019x** [comment]
Oh wow, well that is unfortunate.  Thank you for the quick response, much appreciated!
